pkg/saga/statemachine/statelang/parser/statemachine_parser.go

```diff
@@ -148,18 +148,23 @@ func (b BaseStateParser) GetIntOrDefault(stateName string, stateMap map[string]i
 		return defaultValue, nil
 	}
 
-	// just use float64 to convert, json reader will read all number as float64
-	valueAsFloat64, ok := value.(float64)
-	if !ok {
+	// use float64 conversion when the configuration file is json, and use int conversion when the configuration file is yaml
```

Comment:
Why is it designed this way

Reply:
Because" encoding/json" The package converts all values to the float64 type when parsing json, but the "FLOAT64" is used. yaml.v3" When the package parses yaml, it will correspond to the float64 and int types respectively, the changes here are mainly for compatibility with the two formats, otherwise some errors will occur.

Reply:
Because "encoding/json" package will convert all numerical values into float64 types when parsing json, but" yaml.v3" package will correspond to float64 and int types when parsing YAML, and the modification here is mainly to be compatible with the two formats, otherwise some errors will occur.